index.js

```diff
@@ -12,12 +12,31 @@ module.exports = function(content) {
   var config = loaderUtils.getLoaderConfig(this, "imageWebpackLoader");
   var options = {
     bypassOnDebug: config.bypassOnDebug || false,
+    interlaced: config.interlaced || false,
+    optimizationLevel: config.optimizationLevel || 3,
     gifsicle: config.gifsicle || {},
     mozjpeg: config.mozjpeg || {},
     pngquant: config.pngquant || {},
     optipng: config.optipng || {},
     svgo: config.svgo || {}
   };
+  // Remove line 15-16 and 23-39 in new major version
+  if(config.gifsicle === undefined){
```

Comment:
Hmm, I think that's kind of out of the scope. The point now is that you can pass directly a set of options to the underlying libraries. So we deprecate passing specific options. Then the validation of those options is a concern of the underlying libraries.